src/amp-story-player/amp-story-player-impl.js

```diff
@@ -349,6 +350,93 @@ export class AmpStoryPlayer {
     this.isLaidOut_ = true;
   }
 
+  /**
+   * Shows the story provided by the URL in the player.
+   * @param {string} storyUrl
+   */
+  show(storyUrl) {
+    const storyIdx = this.stories_.findIndex(({href}) => href === storyUrl);
+
+    // TODO(Enriqe): replace for add() once implemented.
+    if (!this.stories_[storyIdx]) {
+      throw new Error(`Story URL not found in the player: ${storyUrl}`);
+    }
+
+    if (storyIdx === this.currentIdx_) {
+      return;
+    }
+
+    this.currentIdx_ = storyIdx;
+
+    this.evictStoriesFromIframes_();
```

Comment:
What if you want to show a story that was already in the iframe pool (but not the current story)?  Won't this cause us to re-render it when we could have just shown it immediately?

Reply:
Yeah, this is an optimization that could come up later. We could just use the internal `next_()` / `previous_()` methods (calling them twice if iframe is > 1 distance away).